test/unit/controller/ResultControllerTest.php

```diff
@@ -20,288 +21,175 @@
-        $basicResppnseSerializerMock->expects($this->once())
+        $this->basicResponseSerializerMock
             ->method('toXml')
-            ->with($this->callback(function (LisOutcomeResponseInterface $response) {
-                return
-                    $response->getStatus() === LisOutcomeResponseInterface::STATUS_INVALID_REQUEST &&
-                    $response->getCodeMajor() === LisOutcomeResponseInterface::CODE_MAJOR_FAILURE;
-            }))
-            ->willReturn('xml_out');
-
-        /** @var ResponseInterface|MockObject $responseMock */
-        $responseMock = $this->createMock(ResponseInterface::class);
-        $responseMock->method('withStatus')->with(400)->willReturn($responseMock);
-        $responseMock->method('withHeader')->with('Content-Type', 'application/xml')->willReturn($responseMock);
-        $responseMock->method('withBody')
-            ->with($this->callback(function (StreamInterface $stream) {
-                return 'xml_out' === (string) $stream;
-            }))
-            ->willReturn($responseMock);
-
-        /** @var LisAuthAdapter|MockObject $listAuthAdapterMock */
-        $listAuthAdapterMock = $this->createMock(LisAuthAdapter::class);
-        $listAuthAdapterMock->method('authenticate')->willReturn($ltiProviderUserMock);
-
-        /** @var LisAuthAdapterFactory|MockObject LisAuthAdapterFactory */
-        $lisAuthAdapterFactory = $this->createMock(LisAuthAdapterFactory::class);
-        $lisAuthAdapterFactory->method('create')->willReturn($listAuthAdapterMock);
-
-        /** @var ServerRequestInterface|MockObject $requestMock */
-        $requestMock = $this->createMock(ServerRequestInterface::class);
-        $requestMock->method('getMethod')->willReturn('POST');
-        $requestMock->method('getBody')->willReturn('request_body');
-
-        $controller = new ResultController();
-        $controller->setServiceLocator($this->getServiceLocatorMock([
-            LisAuthAdapterFactory::class => $lisAuthAdapterFactory,
-            LisOutcomeRequestParser::class => $lisOutcomeRequestParserMock,
-            BasicResponseSerializer::class => $basicResppnseSerializerMock
-        ]));
-        $controller->setRequest($requestMock);
-        $controller->setResponse($responseMock);
-        $controller->manageResults();
-        $response = $controller->getPsrResponse();
-        $this->assertSame($responseMock, $response);
+            ->willReturn('some string');
+
+        $this->responseMock
+            ->expects($this->once())
+            ->method('withStatus')
+            ->with(StatusCode::HTTP_INTERNAL_SERVER_ERROR)
+            ->willReturn($this->responseMock);
+
+        $this->responseMock
+            ->expects($this->once())
+            ->method('withBody')
+            ->willReturn($this->responseMock);
+
+        $this->subject->manageResults();
+    }
+
+
+    public function testManageResultsGET(): void
```

Comment:
Why http method in the test? Is there any different behavior for other http methods?

Reply:
I' testing non POST request

Reply:
so the method should be `testManageResultsWithGetIsNotAllowed`